diff --git a/gfx/skia/skia/include/private/base/SkFeatures.h b/gfx/skia/skia/include/private/base/SkFeatures.h
--- a/gfx/skia/skia/include/private/base/SkFeatures.h
+++ b/gfx/skia/skia/include/private/base/SkFeatures.h
@@ -111,7 +111,8 @@
     #if defined(__AVX512F__) && defined(__AVX512DQ__) && defined(__AVX512CD__) && \
         defined(__AVX512BW__) && defined(__AVX512VL__)
         #define SK_CPU_X64_LEVEL    SK_CPU_X64_LEVEL_ML4
-    #elif defined(__AVX2__)
+    #elif defined(__AVX2__) && defined(__BMI__) && defined(__BMI2__) && \
+          defined(__F16C__) && defined(__FMA__)
         #define SK_CPU_X64_LEVEL    SK_CPU_X64_LEVEL_AVX2
     #elif defined(__AVX__)
         #define SK_CPU_X64_LEVEL    SK_CPU_X64_LEVEL_AVX
@@ -143,7 +144,8 @@
     #if defined(__AVX512F__) && defined(__AVX512DQ__) && defined(__AVX512CD__) && \
         defined(__AVX512BW__) && defined(__AVX512VL__)
         #define SK_CPU_X64_LEVEL        SK_CPU_X64_LEVEL_ML4
-    #elif defined(__AVX2__)
+    #elif defined(__AVX2__) && defined(__BMI__) && defined(__BMI2__) && \
+          defined(__F16C__) && defined(__FMA__)
         #define SK_CPU_X64_LEVEL        SK_CPU_X64_LEVEL_AVX2
     #elif defined(__AVX__)
         #define SK_CPU_X64_LEVEL        SK_CPU_X64_LEVEL_AVX
